### PR TITLE
Update getServerSideProps docs with req.cookie note

### PR DESCRIPTION
**Pull request by vitorbal**
## Documentation / Examples

Add a note to the `getServerSideProps` docs that `context.req` [contains](https://github.com/vercel/next.js/blob/c27e3a41dcf9590fb1c9ba9b9e16517cf4cb49b4/packages/next/types/index.d.ts#L148-L150) a `context.req.cookies` attribute.

I based the copy off of the one from https://nextjs.org/docs/api-routes/api-middlewares.

Checklist:
- [x] Make sure the linting passes

### CODE REVIEW

File: docs/basic-features/data-fetching.md [hunk 1]
```diff
@@ -660,7 +660,7 @@ export async function getServerSideProps(context) {
 The `context` parameter is an object containing the following keys:
 
 - `params`: If this page uses a dynamic route, `params` contains the route parameters. If the page name is `[id].js` , then `params` will look like `{ id: ... }`. To learn more, take a look at the [Dynamic Routing documentation](/docs/routing/dynamic-routes.md).
-- `req`: [The HTTP IncomingMessage object](https://nodejs.org/api/http.html#http_class_http_incomingmessage).
+- `req`: [The HTTP IncomingMessage object](https://nodejs.org/api/http.html#http_class_http_incomingmessage), with an additional `req.cookies` attribute – an object containing the cookies sent by the request (defaults to `{}`).
```
Comment (styfle) [inline docs/basic-features/data-fetching.md:663]: I think we need a more formal definition here.

For example, [API routes](
https://nextjs.org/docs/api-routes/introduction) uses the following:

- req: An instance of http.IncomingMessage, plus some [pre-built middlewares](https://nextjs.org/docs/api-routes/api-middlewares)
- res: An instance of http.ServerResponse, plus some [helper functions](https://nextjs.org/docs/api-routes/response-helpers)

cc @leerob @molebox

Comment (vitorbal) [reply docs/basic-features/data-fetching.md:663]: @styfle yeah good call, I was wondering about that too 🤔

File: docs/basic-features/data-fetching.md [hunk 2]
```diff
@@ -727,6 +727,12 @@ The `context` parameter is an object containing the following keys:
 >
 > Fetching from an external API is fine!
 
+### Built-in `context.req` parsing
+
+To improve the developer experience, the `context.req` request object received by `getServerSideProps` is pre-parsed to include some additional helpful properties. Those properties are:
+
+- `context.req.cookies` - An object containing the cookies sent by the request. Defaults to `{}`
```
Comment (vitorbal) [inline docs/basic-features/data-fetching.md:734]: @styfle  @leerob @molebox took another stab at this, with a dedicated sub-section for `context.req` built-in parsing. What do you all think?

Comment (styfle) [reply docs/basic-features/data-fetching.md:734]: Hmm, I was thinking there were more helpers than just `cookies` but then I checked the type and its just `cookies` like you mentioned 

https://github.com/vercel/next.js/blob/c27e3a41dcf9590fb1c9ba9b9e16517cf4cb49b4/packages/next/types/index.d.ts#L148-L151

And it looks like `res` doesn't have any helpers so maybe its best to go back to your original commit where it was mentioned inline.

Comment (styfle) [reply docs/basic-features/data-fetching.md:734]: Or maybe we discovered a strange behavior where `getServerSideProps()` differs to much from API Routes and we should try to use the same helpers for both 🤔

File: docs/basic-features/data-fetching.md [hunk 3]
```diff
@@ -660,7 +660,7 @@ export async function getServerSideProps(context) {
 The `context` parameter is an object containing the following keys:
 
 - `params`: If this page uses a dynamic route, `params` contains the route parameters. If the page name is `[id].js` , then `params` will look like `{ id: ... }`. To learn more, take a look at the [Dynamic Routing documentation](/docs/routing/dynamic-routes.md).
-- `req`: [The HTTP IncomingMessage object](https://nodejs.org/api/http.html#http_class_http_incomingmessage), with an additional `req.cookies` attribute – an object containing the cookies sent by the request (defaults to `{}`).
+- `req`: [The HTTP IncomingMessage object](https://nodejs.org/api/http.html#http_class_http_incomingmessage), plus [some built-in parsing](#built-in-contextreq-parsing).
```
Comment (molebox) [inline docs/basic-features/data-fetching.md:663]: Lets try to be more explicit when we are describing things and linking to other places.

How about:
```suggestion
- `req`: [The HTTP IncomingMessage object](https://nodejs.org/api/http.html#http_class_http_incomingmessage), plus additional [built-in parsing helpers](#built-in-contextreq-parsing).
```

File: docs/basic-features/data-fetching.md (hunk 2 above)
Comment (molebox) [reply docs/basic-features/data-fetching.md:734]: Personally, I don't think it hurts to have this extra section, even if it just talks about 1 helper. Nice for google foo

Comment (vitorbal) [reply docs/basic-features/data-fetching.md:734]: I agree with @molebox, I like that we'd be able to deeplink to this section. Additionally, this makes it easier to expand the list in the future if we do want to get to parity with the API route helpers

**Review by molebox — DISMISSED** [review]
seal-of-approval

**Comment by vitorbal** [comment]
@molebox do you have the super powers necessary to approve running the workflows?

**Comment by molebox** [comment]
@vitorba
i-got-the-power
l

File: docs/basic-features/data-fetching.md [hunk 4]
```diff
@@ -727,6 +727,12 @@ The `context` parameter is an object containing the following keys:
 >
 > Fetching from an external API is fine!
 
+### Built-in `context.req` parsing
+
+To improve the developer experience, the `context.req` request object received by `getServerSideProps` is pre-parsed to include some additional helpful properties. Those properties are:
```
Comment (ijjk) [inline docs/basic-features/data-fetching.md:732]:
```suggestion
The `req` in the context passed to `getServerSideProps` provides built in middleware that parses the incoming request (req). That middleware is:
```

Tweaked the wording a bit to try to match closer to https://nextjs.org/docs/api-routes/api-middlewares

File: docs/basic-features/data-fetching.md [hunk 5]
```diff
@@ -727,6 +727,12 @@ The `context` parameter is an object containing the following keys:
 >
 > Fetching from an external API is fine!
 
+### Built-in `context.req` parsing
```
Comment (ijjk) [inline docs/basic-features/data-fetching.md:730]:
```suggestion
### Provided `req` middleware in `getServerSideProps`
```

File: docs/basic-features/data-fetching.md [hunk 6]
```diff
@@ -660,7 +660,7 @@ export async function getServerSideProps(context) {
 The `context` parameter is an object containing the following keys:
 
 - `params`: If this page uses a dynamic route, `params` contains the route parameters. If the page name is `[id].js` , then `params` will look like `{ id: ... }`. To learn more, take a look at the [Dynamic Routing documentation](/docs/routing/dynamic-routes.md).
-- `req`: [The HTTP IncomingMessage object](https://nodejs.org/api/http.html#http_class_http_incomingmessage).
+- `req`: [The HTTP IncomingMessage object](https://nodejs.org/api/http.html#http_class_http_incomingmessage), plus additional [built-in parsing helpers](#built-in-contextreq-parsing).
```
Comment (ijjk) [inline docs/basic-features/data-fetching.md:663]:
```suggestion
- `req`: [The HTTP IncomingMessage object](https://nodejs.org/api/http.html#http_class_http_incomingmessage), plus additional [built-in parsing helpers](#provided-req-middleware-in-getserversideprops).
```

File: docs/basic-features/data-fetching.md (hunk 2 above)
Comment (ijjk) [inline docs/basic-features/data-fetching.md:734]:
```suggestion
- `req.cookies` - An object containing the cookies sent by the request. Defaults to `{}`
```
Let's remove the `context` prefix here

**Comment by vitorbal** [comment]
@ijjk thanks, good tweaks! Added 😃

**Review by ijjk — APPROVED** [review]
Thanks!

**Pull request merged**